src/main/java/com/efficio/fieldbook/web/common/controller/CrossingSettingsController.java

```diff
@@ -263,6 +272,58 @@ public String getContentName() {
 
 	}
 
+	/**
+	 * Validates the Breeding Methods in the import file
+	 *
+	 * @return a JSON result object
+	 */
+	@ResponseBody
+	@RequestMapping(value = "/validateBreedingMethods/{isBasedOnImportFile}/{breedingMethodId}", method = RequestMethod.GET)
```

Comment:
remove isBasedOnImportFile and pass breedingMethodId as optional request param. So that if breedingMethodId is not supplied, then it validates the methods from file, otherwise, validate the breedingMethodId. Improve description (javadoc) of the method.